src/main/java/graphql/schema/DataFetchingEnvironmentImpl.java

```diff
@@ -205,9 +214,10 @@ public ExecutionStepInfo getExecutionStepInfo() {
         return executionStepInfo.get();
     }
 
+
     @Override
     public <K, V> @Nullable DataLoader<K, V> getDataLoader(String dataLoaderName) {
-        return dataLoaderRegistry.getDataLoader(dataLoaderName);
+        return new DataLoaderWithContext<>(this, dataLoaderName, dataLoaderRegistry.getDataLoader(dataLoaderName));
```

Comment:
This code creates a new object every time its called

If code was to call this N time inside the same DFE then it would create new object each time which seems wasteful

Can we have a map inside the DFE (or a component class that holds that) which remembers which DataLoaderWithContext object it has made.

Perhaps a `DataLoaderWithContextFactory` which is just the wrapper of the map and with enough knowledge to know what represent a proper key (its more than data loader name I suspect)